pkg/build/admission/secretinjector/admission.go

```diff
@@ -105,6 +116,30 @@ func (si *secretInjector) Admit(attr admission.Attributes) (err error) {
 	return nil
 }
 
+func (si *secretInjector) admitSecret(attr admission.Attributes, secret *api.Secret) (err error) {
```

Comment:
can't this be part of the static validation logic for secrets, or that's not an option because it's an upstream object so we can't override/extend the validation logic?